src/@next/components/organisms/CartSidebar/CartSidebar.tsx

```diff
@@ -0,0 +1,225 @@
+import React from "react";
+import { FormattedMessage } from "react-intl";
+import ReactSVG from "react-svg";
+
+import {
+  Button,
+  CartSummaryCosts,
+  Loader,
+  OfflinePlaceholder,
+} from "@components/atoms";
+import { CardHeader } from "@components/molecules";
+import { TaxedMoney } from "@components/containers";
+import { useHandlerWhenClickedOutside, useNetworkStatus } from "@hooks";
+import { ITaxedMoney } from "@types";
+
+import { IItems } from "@saleor/sdk/lib/api/Cart/types";
+
+import cartImg from "images/cart.svg";
+import { CartRow, Overlay } from "..";
+import * as S from "./styles";
+
+const generateCart = (
+  items: IItems,
+  removeItem: (variantId: string) => any,
+  updateItem: (variantId: string, quantity: number) => any
+) => {
+  return items?.map(({ id, variant, quantity, totalPrice }, index) => (
+    <CartRow
+      type="condense"
+      key={id ? `id-${id}` : `idx-${index}`}
+      index={index}
+      id={variant?.product?.id || ""}
+      name={variant?.product?.name || ""}
+      maxQuantity={variant.quantityAvailable || quantity}
+      quantity={quantity}
+      onRemove={() => removeItem(variant.id)}
+      onQuantityChange={quantity => updateItem(variant.id, quantity)}
+      thumbnail={{
+        ...variant?.product?.thumbnail,
+        alt: variant?.product?.thumbnail?.alt || "",
+      }}
+      totalPrice={<TaxedMoney taxedMoney={totalPrice} />}
+      unitPrice={<TaxedMoney taxedMoney={variant?.pricing?.price} />}
+      sku={variant.sku}
+      attributes={variant.attributes?.map(attribute => {
+        return {
+          attribute: {
+            id: attribute.attribute.id,
+            name: attribute.attribute.name || "",
+          },
+          values: attribute.values.map(value => {
+            return {
+              id: value?.id,
+              name: value?.name || "",
+              value: value?.value,
+            };
+          }),
+        };
+      })}
+    />
+  ));
+};
+
+const prepareCartFooter = (
+  totalPrice?: ITaxedMoney | null,
+  shippingTaxedPrice?: ITaxedMoney | null,
+  promoTaxedPrice?: ITaxedMoney | null,
+  subtotalPrice?: ITaxedMoney | null
+) => (
+  <CartSummaryCosts
+    subtotalPrice={
+      <TaxedMoney data-test="subtotalPrice" taxedMoney={subtotalPrice} />
+    }
+    totalPrice={<TaxedMoney data-test="totalPrice" taxedMoney={totalPrice} />}
+    shippingPrice={
+      shippingTaxedPrice &&
+      shippingTaxedPrice.gross.amount !== 0 && (
```

Comment:
Logic of displaying prices (or not) should be handled inside CartSummaryCosts.tsx